build/mergeallmodules.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+
+#Assumes that ./x64 and ./x86 exist and contain the arch-specific copies of a DSC resource module
+
+set -e
+
+zip -L >/dev/null 2>&1
+if [ $? != 0 ]; then
+        echo "zip must be installed"
+        exit 1
+fi
+
+unzip -Z >/dev/null 2>&1
+if [ $? != 0 ]; then
+        echo "unzip must be installed"
+        exit 1
+fi
+
+TargetDir="./target"
+x64dir="./x64"
+x86dir="./x86"
+TestSigningDir=$1
+
+if [ "$TestSigningDir" = "" ]
+then
+	TestSigningDir="../test/config/testsigning"
+fi
+
+if [ -d "$TestSigningDir" ]
+then
+	SigningKeyFilePath="$TestSigningDir/signingkeys.gpg"
+	SigningKeyPassphrase="$TestSigningDir/passphrase"
+
+	if [ ! -f "$SigningKeyFilePath" ]
+	then
+		echo "GPG key file is not found"
+		exit 1
+	fi
+
+	if [ ! -f "$SigningKeyPassphrase" ]
+	then
+		echo "GPG key passphrase is not found"
+		exit 1
+	fi
+else
+	echo "Signing key directory doesn't exit. $TestSigningDir"
+        exit 1
+fi
+
+if [ -d "$TargetDir" ]
+then
+	rm -rf "$TargetDir"
+fi
+
+mkdir "$TargetDir"
+
+for ModuleFilePath in x64/*.zip;
+do
+        ModuleFileName=`echo $ModuleFilePath | xargs -n 1 basename`
+	ModuleName=`echo $ModuleFileName | cut -d _ -f 1`
+	Version=`echo $ModuleFileName | cut -d _ -f 2 |cut -d . -f 1,2`
+	
+	if [ -f "$x64dir/$ModuleFileName" ]
+	then
+		echo "Found x64 module .zip: "$x64dir/$ModuleFileName""
+	else
+		echo "Did not file x64 module .zip: "$x64dir/$ModuleFileName""
+		exit 1
+	fi
+	
+	if [ -f "$x86dir/$ModuleFileName" ]
+	then
+		echo "Found x86 module .zip: "$x86dir/$ModuleFileName""
+	else
+		echo "Did not file x86 module .zip: "$x86dir/$ModuleFileName""
+		exit 1
+	fi
+	
+	#Merge modules to ./target/ModuleFileName
+	unzip -o $x86dir/$ModuleFileName -d ./target
+	unzip -o $x64dir/$ModuleFileName -d ./target
+	
+	WorkingDir="./target/$ModuleName/"
+	
+	#Generate Sha256sums
+	echo "Generating sha256sums..."
+	cd $WorkingDir
+	find . -type f -print0 | xargs -0 sha256sum |grep -v sha256sums > ./$ModuleName.sha256sums
+	
+	#Test Sha256sums
+	echo "Test sha256sums..."
+	sha256sum -c $ModuleName.sha256sums
+	cd ../../
+
+	#Test sign module
+	echo "signing .sha256sums file..."
+	cat $SigningKeyPassphrase | gpg --no-default-keyring --keyring $SigningKeyFilePath --yes --passphrase-fd 0 --output ./$WorkingDir/$ModuleName.asc --armor --detach-sign ./$WorkingDir/$ModuleName.sha256sums
+	if [ $? != 0 ]; then
+		echo "Failed to sign the .sha256sums file"
+		exit 1
+	fi
+
+	echo "verifying signature..."
+	gpg --no-default-keyring --keyring $SigningKeyFilePath --verify  ./$WorkingDir/$ModuleName.asc ./$WorkingDir/$ModuleName.sha256sums
+	if [ $? != 0 ]; then
+		echo "signture is invalid"
+		exit 1
+	fi
+
+	echo "Producing .zip file"
+	(
+		cd ./target
+		zip -r $ModuleName.zip $ModuleName/*
+	)
+done
+
+cp ./target/*.zip ../target/
```

Comment:
So what it is, EXACTLY, that's getting published. What are you after here, the "real" target directory, at one level up from the architecture specific directories. It sounds like "./target" isn't really a target directory, it's a working directory. I strongly recommend that, rather than creating new directories with confusion of what's what, you modify this to meet our existing standards: Temporary directories go into "intermediate" and files to stage go into "target" (based from omsagent directory).

Otherwise, this is just needlessly confusing. I'm not even certain what to stage to the build directories at this point.